lib/buildkite/collector/socket_connection.rb

```diff
@@ -76,6 +76,9 @@ def initialize(session, url, headers)
             @session.handle(self, data.data)
           end
         end
+      # These get re-raise from session, we should fail gracefully
+      rescue Buildkite::Collector::Session::DISCONNECTED_EXCEPTIONS => e
```

Comment:
I agree this is the right thing to do, however I'm not sure if we need to attempt to execute the disconnect flow. Thoughts?

Reply:
> if we need to attempt to execute the disconnect flow.

??? Here we rescue the exception raised from `Session`, only logs an error, did not execution the disconnect flow.

This is attempt execute disconnect flow looks like in the `rescue IndexError` block:

https://github.com/buildkite/rspec-buildkite-analytics/blob/fbce8caa2b535a4cf1098222026abb045d569e66/lib/buildkite/collector/socket_connection.rb#L134-L138

Am I missing something 🤔 😅

Reply:
@JuanitoFatas I couldn't really follow what you're saying.

@blaknite if I'm understanding it correctly I don't think we want to execute the disconnect flow. The exceptions it's catching are "DISCONNECTED_EXCEPTIONS", i.e., we have already tried to perform the disconnect flow and it raised an error. Now we should print an error and gracefully handle the exception.